tests/ha/check_hae_active.py

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Check that SUT has the HA Extension Enabled. If it's not, soft fail and work around
+# Maintainer: qe-sap@suse.com
+
+from testapi import *
+import json
+
+perl.require("registration")
+perl.require("serial_terminal")
+perl.require("utils")
+perl.require("suseconnect_register")
+
+def check_suseconnect(self):
+    '''
+    Check that SUT has suseconnect-ng installed. Install it if it's not there
+    '''
+    no_suseconnect = int(script_run("which SUSEConnect"))
+    if (no_suseconnect):
+        record_soft_failure("bsc#1238913 - SUSEConnect is not installed by default")
+        perl.utils.zypper_call("in suseconnect-ng")
+
+def check_product_registration(self, status=""):
+    '''
+    Check SUT is correctly registered to the SCC proxy. Re-register if it's not registered to
+    the proxy, if it's not registered or if it's missing /etc/SUSEConnect
+    '''
+    # Retval 0 means file exists
+    etc_suseconnect_exists = (0 == int(script_run("test -f /etc/SUSEConnect")))
+    scc_proxy_reg = False
+    scc_url = get_var("SCC_URL", f"http://all-{get_required_var('BUILD')}.proxy.scc.suse.de")
+
+    if (etc_suseconnect_exists):
+        suseconnect_config = script_output("cat /etc/SUSEConnect")
+        record_info("/etc/SUSEConnect", suseconnect_config)
+        scc_proxy_reg = any(scc_url in _ for _ in suseconnect_config.split("\n"))
+
+    scc_status = json.loads(status)
+    
+    if ((not etc_suseconnect_exists) or (scc_status[0]["status"] == "Not Registered") or (not scc_proxy_reg)):
```

Comment:
What is `0` in `scc_status[0]` ? Is it about the fact that json string coming from the external (that is `SUSEConnect --status` output) is a list and **first** element is always the registration status of the image?

Reply:
Got a reply, and there's no guarantee the first item refers to the main product. I'll change this before merging.